aptos-move/framework/aptos-names/sources/verify.move

```diff
@@ -0,0 +1,35 @@
+module aptos_names::verify {
+    use std::string;
+    use std::vector;
+    use aptos_framework::chain_id;
+    use aptos_std::ed25519;
+    use aptos_names::config;
+
+    friend aptos_names::domains;
+
+    struct RegisterDomainProofChallenge has drop {
+        sequence_number: u64,
+        register_address: address,
+        domain_name: string::String,
+        chain_id: u8,
+    }
+
+    const EINVALID_PROOF_OF_KNOWLEDGE: u64 = 1;
+
+    public(friend) fun verify_register_domain_signature(signature: vector<u8>, sequence_number: u64, account_address: address, domain_name: string::String) {
+        let chain_id = chain_id::get();
+        let register_domain_proof_challenge = RegisterDomainProofChallenge {
+            sequence_number,
+            register_address: account_address,
+            domain_name,
+            chain_id
+        };
+
+        let admin_public_key_bytes = config::admin_public_key();
+        // the number at index 0 is the size of the vector, so we need to remove it to get a valid 32-byte public key
```

Comment:
I don't like this: why is `admin_public_key` not returning the actual public key set via `set_admin_public_key`?

Is this because `read_u8_vector_v1` doesn't return the vector itself, but a BCS serialization of the vector, which includes its length?

If so, this length should be discarded in in `read_u8_vector_v1`... Otherwise, this API design will bite us later.

@davidiw, not trying to be nitpicky 😄, but we need to have a clean API here...

Reply:
ahh yeah solid point, just moved the code that discards the length to `read_u8_vector_v1`

Reply:
Otherwise, just hit "Resolve conversation" 😄